pkg/ruler/api.go

```diff
@@ -268,3 +282,274 @@ func (r *Ruler) alerts(w http.ResponseWriter, req *http.Request) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (r *Ruler) getRuleGroup(w http.ResponseWriter, req *http.Request) {
+	logger := util.WithContext(req.Context(), util.Logger)
+	userID, err := user.ExtractOrgID(req.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	vars := mux.Vars(req)
+
+	// Parse the namespace from the url path parameters and return
+	// a 400 if it is invalid
+	namespace, err := parseNamespace(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Parse the rule group name from the url path parameters and return
+	// a 400 if it is invalid
+	groupName, err := parseGroupName(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	rg, err := r.store.GetRuleGroup(req.Context(), userID, namespace, groupName)
+	if err != nil {
+		if err == store.ErrGroupNotFound {
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		}
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	formattedRG := store.FromProto(rg)
+
+	d, err := yaml.Marshal(&formattedRG)
+	if err != nil {
+		level.Error(logger).Log("msg", "error marshalling yaml rule groups", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/yaml")
+	if _, err := w.Write(d); err != nil {
+		level.Error(logger).Log("msg", "error writing yaml response", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (r *Ruler) createRuleGroup(w http.ResponseWriter, req *http.Request) {
+	logger := util.WithContext(req.Context(), util.Logger)
+	userID, err := user.ExtractOrgID(req.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Parse the namespace from the url path parameters and return
+	// a 400 if it is invalid
+	namespace, err := parseNamespace(mux.Vars(req))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	payload, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		level.Error(logger).Log("err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	level.Debug(logger).Log("msg", "attempting to unmarshal rulegroup", "userID", userID, "group", string(payload))
+
+	rg := rulefmt.RuleGroup{}
+	err = yaml.Unmarshal(payload, &rg)
+	if err != nil {
+		level.Error(logger).Log("err", err.Error())
+		http.Error(w, ErrBadRuleGroup.Error(), http.StatusBadRequest)
+		return
+	}
+
+	errs := ValidateRuleGroup(rg)
+	if len(errs) > 0 {
+		level.Error(logger).Log("err", err.Error())
+		http.Error(w, errs[0].Error(), http.StatusBadRequest)
+		return
+	}
+
+	rgProto := store.ToProto(userID, namespace, rg)
+
+	level.Debug(logger).Log("msg", "attempting to store rulegroup", "userID", userID, "group", rgProto.String())
```

Comment:
`rgProto.String()` can be huge here, do we really want to log it? Same for logging payload above.

Reply:
Since this log is debug level, I think it should be ok?